pipeline/test/services/service-cluster/testOpensearch.sh

```diff
@@ -0,0 +1,431 @@
+            fields_limit_usage=$((fields_count * 100 / fields_limit))
+            if [[ $fields_count -gt $fields_limit ]]; then
+                no_error=false
+            fi
+            if [[ $fields_limit_usage -gt 50 ]]; then
+                warn=true
+                debug_msg+="${index}\t\t ${fields_count} \t\t ${fields_limit} \n"
+            fi
+        fi
+    done
+
+    if $no_error; then
+        echo "success ✔"
+        echo "[DEBUG] Fields limit has not been reached yet"
+        if $warn; then
+            echo -e "$debug_msg"
+        fi
+    else
+        echo "failure ❌"
+        echo "[ERROR] Some fields limit have been crossed"
+        echo -e "$debug_msg"
+    fi
+}
+
+check_opensearch_user_roles() {
+    echo -ne "checking user roles mappings ... "
+    no_error=true
+    debug_msg=""
+
+    readarray configuredMappings < <(yq4 e -o=j -I=0 '.opensearch.extraRoleMappings[]' "${config['config_file_sc']}")
+
+    rolesmapping=$(curl -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_plugins/_security/api/rolesmapping")
+
+    for roleMapping in "${configuredMappings[@]}"; do
+        configured_mapping_name=$(echo "$roleMapping" | yq4 e '.mapping_name' -)
+        configured_users=$(echo "$roleMapping" | yq4 e '.definition.users[]' -)
+        res=$(echo "$rolesmapping" | jq ".$configured_mapping_name")
+        if [[ $res != "null" ]]; then
+            users=$(echo "$rolesmapping" | jq -r ".$configured_mapping_name.users[]")
+            if [[ "$users" != "$configured_users" ]]; then
+                no_error=false
+                debug_msg+="[ERROR] Missing users < ${configured_users//$'\n'/ }> != < $users >, for role mapping: $configured_mapping_name \n"
+            fi
+        else
+            no_error=false
+            debug_msg+="[ERROR] Missing role mapping_name ${roleMapping} \n"
+        fi
+    done
+
+    if $no_error; then
+        echo "success ✔"
+        echo "[DEBUG] User roles are configured correctly"
+    else
+        echo "failure ❌"
+        echo -ne "$debug_msg"
+    fi
+}
+
+check_opensearch_ism() {
+    echo -ne "Checking opensearch Index State Management ... "
+    no_error=true
+    debug_msg=""
+
+    default_policies_enabled=$(yq4 -e '.opensearch.ism.defaultPolicies' "${config['config_file_sc']}")
+    if [[ $default_policies_enabled == "true" ]]; then
+        default_policies=("kubernetes" "kubeaudit" "authlog" "other")
+        for policy in "${default_policies[@]}"; do
+            res=$(curl -w "%{http_code}" -o /dev/null -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_plugins/_ism/policies/${policy}")
+            if [[ $res != "200" ]]; then
+                no_error=false
+                debug_msg+="[ERROR] Missing default policy : ${policy}\n"
+            fi
+        done
+    fi
+
+    # additional_policies=$(yq4 -e '.opensearch.ism.additionalPolicies | keys' "${config['config_file_sc']}")
```

Comment:
```suggestion
```
Remove?

Reply:
Yep, thanks a lot, I forgot to do the comments cleanup